src/sampler/status.jl

```diff
@@ -19,32 +19,24 @@ If a generator is inactive:
 3. Construct the status vector to be all ones except for zero at the sampled generator.
 """
 struct Nminus1StatusSampler <: AbstractStatusSampler
-    data::Dict
+    E::Int
+    G::Int
+    non_bridges::Vector{Int}
 end
 
 function Random.rand(rng::AbstractRNG, rs::Nminus1StatusSampler)
     p = rand(rng)
 
-    E = length(rs.data["branch"])
-    G = length(rs.data["gen"])
-
-    br_status = ones(Int, E)
-    gen_status = ones(Int, G)
+    br_status = ones(Int, rs.E)
+    gen_status = ones(Int, rs.G)
```

Comment:
If you want zero-one valued vector, use a boolean-typed vector
```suggestion
    br_status = ones(Bool, rs.E)
    gen_status = ones(Bool, rs.G)
```

Reply:
Note that you could also return the index of the generator / branch being disabled, which would save some memory